internal/impl/kafka/input_kafka_franz.go

```diff
@@ -263,10 +281,18 @@ func (f *franzKafkaReader) Connect(ctx context.Context) error {
 
 	checkpoints := newCheckpointTracker()
 
+	var initialOffset kgo.Offset
+	if f.startFromOldest {
+		initialOffset = kgo.NewOffset().AtStart()
+	} else {
+		initialOffset = kgo.NewOffset().AtEnd()
+	}
+
 	clientOpts := []kgo.Opt{
 		kgo.SeedBrokers(f.seedBrokers...),
 		kgo.ConsumerGroup(f.consumerGroup),
 		kgo.ConsumeTopics(f.topics...),
+		kgo.ConsumeResetOffset(initialOffset),
```

Comment:
Seems to be the advised way to do this:

https://github.com/twmb/franz-go/issues/42#issuecomment-1155746452

Reply:
Might want to try the AfterMilli option